### PR TITLE
PILOT-1488: Create view to GET single dataset

### PR DESCRIPTION
**Pull request by hugoantunes**
## Summary

Get single dataset by code or by id. 

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1488

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [x] Yes
- [ ] No

## Test Directions

(Additional instructions for how to run tests or validate functionality if not covered by unit tests)

### CODE REVIEW

File: app/components/dataset/crud.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from uuid import UUID
+
+from sqlalchemy.future import select
+
+from app.components.crud import CRUD
+from app.models import Dataset
+
+
+class DatasetCRUD(CRUD):
+    """CRUD for managing project database models."""
```
Comment (vmoshynskyi) [inline app/components/dataset/crud.py:26]:
```suggestion
    """CRUD for managing dataset database models."""
```

File: app/components/dataset/crud.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    model = Dataset
+
+    async def retrieve_by_code(self, code: str) -> Dataset:
+        """Get an existing project by unique code."""
```
Comment (vmoshynskyi) [inline app/components/dataset/crud.py:31]:
```suggestion
        """Get an existing dataset by unique code."""
```

File: app/components/dataset/crud.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        statement = select(self.model).where(self.model.code == code)
+
+        project = await self._retrieve_one(statement)
+
+        return project
+
+    async def retrieve_by_id_or_code(self, id_or_code: Union[UUID, str]) -> Dataset:
+        """Get an existing project either by id or by code (depending on type)."""
```
Comment (vmoshynskyi) [inline app/components/dataset/crud.py:40]:
```suggestion
        """Get an existing dataset either by id or by code (depending on type)."""
```

File: app/components/filtering.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+from typing import Type
+
+from pydantic import BaseModel
+from sqlalchemy.sql import Select
+
+from app.components import DBModel
+
+
+class Filtering(BaseModel):
```
Comment (vmoshynskyi) [inline app/components/filtering.py:24]: I suggest to copy also tests for this class.

File: app/components/models.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Any
+from typing import Optional
+
+from app.components.db_model import DBModel
+
+
+class ModelList(list):
```
Comment (vmoshynskyi) [inline app/components/models.py:22]: I suggest to copy also tests for this class.

File: app/components/parameters.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+    def to_pagination(self) -> Pagination:
+        return Pagination(page=self.page, page_size=self.page_size)
+
+
+class SortByFields(StrEnum):
+    """Base class for defining sort by fields."""
+
+
+class SortParameters(QueryParameters):
```
Comment (vmoshynskyi) [inline app/components/parameters.py:50]: I suggest to copy also tests for this class.

File: app/components/sorting.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+
+class SortingOrder(StrEnum):
+    """Available sorting orders."""
+
+    ASC = 'asc'
+    DESC = 'desc'
+
+
+class Sorting(BaseModel):
```
Comment (vmoshynskyi) [inline app/components/sorting.py:35]: I suggest to copy also tests for this class.

File: app/components/types.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from enum import Enum
+
+
+class StrEnum(str, Enum):
```
Comment (vmoshynskyi) [inline app/components/types.py:19]: I suggest to copy also tests for this class.

**Review by vmoshynskyi — APPROVED** [review]
Looks fine. Please check few comments and suggestions.

File: app/components/filtering.py (hunk 4 above)
Comment (hugoantunes) [reply app/components/filtering.py:24]: I will add in the next PR :D

File: app/components/models.py (hunk 5 above)
Comment (hugoantunes) [reply app/components/models.py:22]: I will add in the next PR :D

File: app/components/parameters.py (hunk 6 above)
Comment (hugoantunes) [reply app/components/parameters.py:50]: I will add in the next PR :D

File: app/components/sorting.py (hunk 7 above)
Comment (hugoantunes) [reply app/components/sorting.py:35]: I will add in the next PR :D

File: app/components/types.py (hunk 8 above)
Comment (hugoantunes) [reply app/components/types.py:19]: I will add in the next PR :D

**Pull request merged**